mkosi

```diff
@@ -1245,14 +1268,30 @@ def invoke_yum(args: CommandLineArguments, workspace: str, repositories: List[st
     with mount_api_vfs(args, workspace):
         run(cmdline, check=True)
 
-def invoke_dnf_or_yum(args: CommandLineArguments, workspace: str, repositories: List[str], packages: List[str], config_file: str) -> None:
+    if cleanup:
+        # Clean up package manager metadata
+        keep_yum_data = 'yum' in packages
+        keep_rpm_data = keep_yum_data or 'rpm' in packages
+
+        if not keep_yum_data:
+            print_step('Cleaning yum metadata...')
+            remove_glob(root + '/var/lib/yum',
+                        root + '/var/log/yum.*',
+                        root + '/var/cache/yum',
+                        root + '/etc/yum/repos.d',
+                        root + '/etc/yum')
+        if not keep_rpm_data:
+            print_step('Cleaning rpm metadata...')
+            remove_glob(root + '/var/lib/rpm')
```

Comment:
hmm, the code a second time? this should be a function and shared between both callers